Restore basic auth for subscriptions API

fosterseth opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
If service account auth doesn't work, fall back to using basic auth.

This allows users to continue to use basic auth credentials in the client ID / secret fields.
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does. Also please make sure that if this PR has an attached JIRA, put AAP-<number>
in as the first entry for your PR title.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

fosterseth committed:
Restore basic auth for subscriptions API

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

chrismeyersfsu3 commented on awx/main/utils/licensing.py:
```diff
@@ -243,14 +243,23 @@ def validate_rh(self, user, pw):
+                'GET',
+                host,
+                verify=True,
+                timeout=(31, 31),
```
A total less than 30 would be best. 5 for the first param, connect, and 20 for the second.

Envoy timeout is 30 seconds so every timeout we have behind envoy should be less than 30.

fosterseth replied:
good callout, fixed

fosterseth committed:
Total timeout within 30 seconds

Envoy has a timeout of 30 seconds, so
the total timeout should be less than that.

(5, 20) means 5 seconds to connect to server,
20 seconds to start reading data.

Signed-off-by: Seth Foster <fosterbseth@gmail.com>
